Make page_title HTML-escape only once

elias19r opened the pull request:
This wraps the string returned by page_title with escape_once from
ActionView::Helpers::TagHelper to make sure it HTML-escapes only once.

Without it, rendering <%= page_title %> in an ERB template causes it
to be HTML-escaped twice.

https://api.rubyonrails.org/v6.1.0/classes/ActionView/Helpers/TagHelper.html#method-i-escape_once

Issue https://github.com/thoughtbot/flutie/issues/73

elias19r committed:
Make page_title HTML-escape only once

This wraps the string returned by page_title with escape_once from
ActionView::Helpers::TagHelper to make sure it HTML-escapes only once.

Without it, rendering <%= page_title %> in an ERB template causes it
to be HTML-escaped twice.

elias19r commented on lib/flutie/page_title_helper.rb:
```diff
@@ -3,11 +3,13 @@ module PageTitleHelper
+        content_for(page_title.page_title_symbol),
+        page_title.separator,
+        options[:reverse],
+      ).to_s,
```
@neilvcarvalho

> there shouldn't be any backwards compatibility issue in making `flutie` return HTML-safe strings

My first solution was calling `.html_safe` here and it worked, but only for the page title content. To make sure custom `separator` and custom `app_name` are also html-escaped, and once, I think we need to go with `escape_once`

neilvcarvalho replied:
Oooh, that's a great point. I hadn't thought of separators and app names needing to be escaped, as in "Johnson & Johnson > São Paulo Headquarters". TIL `escape_once`.

neilvcarvalho commented on lib/flutie/page_title_helper.rb:
```diff
@@ -3,11 +3,13 @@ module PageTitleHelper
+        page_title.separator,
+        options[:reverse],
+      ).to_s,
+    )
```
```suggestion
    ).html_safe
```

I think it's failing because of this change: https://github.com/rails/rails/pull/48265. It was released on Rails 7.1, so it makes sense that it's passing from that version onwards. Making it `html_safe` should avoid the double-escaping issue on older Rails.

elias19r replied:
Good catch!

elias19r committed:
Apply html_safe to string returned by escape_once

escape_once returns an html-safe string but only in Rails >= 7.1.
So for compatibility with older versions, we need to explicitly
apply html_safe here.

Co-authored-by: Neil Carvalho <me@neil.pro>

neilvcarvalho commented:
Thanks for your contribution!
